src/toil/job.py

```diff
@@ -1722,3 +1721,185 @@ def convertPromises(kwargs):
             elif isinstance(kwargs.get(r), PromisedRequirement):
                 foundPromisedRequirement = True
         return foundPromisedRequirement
+
+
+def docker_call(job,
+                tool,
+                parameters=None,
+                work_dir='.',
+                docker_parameters=None,
+                outfile=None,
+                check_output=False,
```

Comment:
If we're refactoring this code, I propose not having `check_output` as a flag, but instead have two functions. Ideally, the `check_output` version of `docker_call` would match up with `check_output`'s return signature.